platform/linux-generic/odp_random_std.c

```diff
@@ -40,38 +41,35 @@ static int32_t _random_data(uint8_t *buf, uint32_t len, uint64_t *seed)
 {
 	const uint32_t ret = len;
 
-	if (!_ODP_UNALIGNED) {
-		uint32_t r = xorshift64s32(seed);
+	if (!_ODP_UNALIGNED && (uintptr_t)buf & 3) {
+		for (uint32_t i = 0; i < len / 4; i++) {
+			uint32_t r = xorshift64s32(seed);
 
-		if (((uintptr_t)buf & 1) && len) {
-			*(uint8_t *)buf = r & 0xff;
-			r >>= 8;
-			buf += 1;
-			len -= 1;
+			memcpy(buf, &r, 4);
```

Comment:
This memcpy does not appear to get optimized away when compiling with GCC 12 for armv6+fp with -mno-unaligned-access. So if we do want to optimize for the _ODP_UNALIGNED == 0 case, the memcpy should perhaps be replaced by direct assignments to buf.